src/actions/BitPayActions.js

```diff
@@ -90,26 +90,26 @@ export async function launchBitPay(
 
   // Select payment wallet, if this wasn't called directly from a wallet's
   // send scene
-  let selectedWallet, currencyCode
+  let selectedWallet, selectedCurrencyCode
   if (params.wallet) {
     // Ensure the core wallet is accepted by this invoice as a payment option
-    currencyCode = params.wallet.currencyInfo.currencyCode
-    if (!paymentCurrencies.includes(currencyCode)) {
+    selectedCurrencyCode = params.wallet.currencyInfo.currencyCode
+    if (!paymentCurrencies.includes(selectedCurrencyCode)) {
       throw new BitPayError(BitPayErrorCode.InvalidPaymentOption, { text: paymentCurrencies.join(', ') })
     }
     selectedWallet = params.wallet
   } else {
     // Check if user owns any wallets that are accepted by the invoice
-    const currencyWalletCodes = Object.keys(params.currencyWallets ?? {}).map(id =>
-      params.currencyWallets ? params.currencyWallets[id].currencyInfo.currencyCode : {}
-    )
-    if (currencyWalletCodes.length === 0 || paymentCurrencies.filter(currencyCode => currencyWalletCodes.includes(currencyCode))) {
+    const { currencyWallets = {} } = params
+    const matchingWallets = Object.keys(currencyWallets).filter(key => paymentCurrencies.includes(currencyWallets[key].currencyInfo.currencyCode))
+    if (matchingWallets === 0) {
       throw new BitPayError(BitPayErrorCode.NoPaymentOption, { text: paymentCurrencies.join(', ') })
     } else {
       const walletListResult = await Airship.show(bridge => (
         <WalletListModal bridge={bridge} headerTitle={s.strings.select_wallet} allowedCurrencyCodes={paymentCurrencies} />
       ))
       const { walletId, currencyCode } = walletListResult
```

Comment:
This destructuring was not setting currencyCode with that from the walletListResult